### PR TITLE
fix(build-and-test-with-yarn): Allow combining src and test files in …

### PR DESCRIPTION
**Pull request by RappC**
…single folder

### CODE REVIEW

File: build-and-test-with-yarn/action.yaml [hunk 1]
```diff
@@ -24,13 +24,25 @@ inputs:
...
+    default: "**/*.spec.ts"
     required: false
   sonar-tests:
     description: Holds a comma-delimited list of directories (not a path matching pattern) relative to the current working directory.
-    default: test/
+    default: src/
+    required: false
+  sonar-test-inclusions:
+    description: One or more path matching patterns defining files to be included in analysis, among those found below directories specified in `sonar.tests`. Applied after `sonar.test.exclusions` and therefore overriding it.
+    default: "**/*.spec.ts"
```
Comment (TheDome) [inline build-and-test-with-yarn/action.yaml:41]: Dont we use test.ts?
```suggestion
    default: "**/*.test.ts"
```

File: build-and-test-with-yarn/action.yaml [hunk 2]
```diff
@@ -24,13 +24,25 @@ inputs:
...
-  sonar-test-exclusions:
-    description: One or more path matching patterns defining files to be excluded from analysis, among those found below the directories specified in `sonar.tests`.
-    default: test/**
+  sonar-inclusions:
+    description: One or more path matching patterns defining files to be included in analysis, among those found below directories specified in `sonar.sources`.
+    default: "**/*.ts"
+    required: false
+  sonar-exclusions:
+    description: One or more path matching patterns defining files to be excluded from analysis, among those found below the directories specified in `sonar.sources`.
+    default: "**/*.spec.ts"
```
Comment (TheDome) [inline build-and-test-with-yarn/action.yaml:33]: Dont we use schema test.ts?
```suggestion
    default: "**/*.test.ts"
```

File: build-and-test-with-yarn/action.yaml [hunk 3]
```diff
@@ -24,13 +24,25 @@ inputs:
...
     required: false
-  sonar-test-exclusions:
-    description: One or more path matching patterns defining files to be excluded from analysis, among those found below the directories specified in `sonar.tests`.
-    default: test/**
+  sonar-inclusions:
+    description: One or more path matching patterns defining files to be included in analysis, among those found below directories specified in `sonar.sources`.
+    default: "**/*.ts"
+    required: false
+  sonar-exclusions:
+    description: One or more path matching patterns defining files to be excluded from analysis, among those found below the directories specified in `sonar.sources`.
```
Comment (TheDome) [inline build-and-test-with-yarn/action.yaml:32]: By what are these separated? comma or semicolon?

**Review by TheDome — CHANGES_REQUESTED** [review]

File: build-and-test-with-yarn/action.yaml (hunk 1 above)
Comment (RappC) [reply build-and-test-with-yarn/action.yaml:41]: The default for nest.js is `spec` and I remember this form other node.js apps.

File: build-and-test-with-yarn/action.yaml (hunk 2 above)
Comment (RappC) [reply build-and-test-with-yarn/action.yaml:33]: see comment above.

File: build-and-test-with-yarn/action.yaml (hunk 1 above)
Comment (RappC) [reply build-and-test-with-yarn/action.yaml]: As agreed during the Daily I have amended this to match `test.ts`

File: build-and-test-with-yarn/action.yaml (hunk 2 above)
Comment (RappC) [reply build-and-test-with-yarn/action.yaml]: As agreed during the Daily I have amended this to match `test.ts`

**Review by TheDome — APPROVED** [review]

**Pull request merged**